### PR TITLE
test: Use WebDriverManager to manage chromedriver

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

File: flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java [hunk 1]
```diff
@@ -59,93 +47,23 @@ private ChromeDriverLocator() {
...
-                                    + " (currently set to "
-                                    + chromedriverProperty
-                                    + ") or available in your PATH");
-                }
-            } else {
+            // This sets the same property
+            WebDriverManager.chromedriver().setup();
+        } else {
+            String location = System.getProperty(WEBDRIVER_CHROME_DRIVER);
+            if (location != null) {
```
Comment (caalador) [inline flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java:62]: Should just be `} else if (chromedriverProperty != null) {` and then log state.

File: flow-tests/test-application-theme/test-reusable-as-parent/pom.xml [hunk 2]
```diff
@@ -63,38 +61,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-application-theme/test-reusable-as-parent/pom.xml:64]:
```suggestion
</project>

```

File: flow-tests/test-application-theme/test-theme-component-live-reload/pom.xml [hunk 3]
```diff
@@ -65,37 +63,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-application-theme/test-theme-component-live-reload/pom.xml:66]:
```suggestion
</project>

```

File: flow-tests/test-application-theme/test-theme-live-reload/pom.xml [hunk 4]
```diff
@@ -59,37 +57,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-application-theme/test-theme-live-reload/pom.xml:60]:
```suggestion
</project>

```

File: flow-tests/test-application-theme/test-theme-reusable/pom.xml [hunk 5]
```diff
@@ -63,38 +61,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-application-theme/test-theme-reusable/pom.xml:64]:
```suggestion
</project>

```

File: flow-tests/test-application-theme/test-theme-switch-live-reload/pom.xml [hunk 6]
```diff
@@ -72,37 +70,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-application-theme/test-theme-switch-live-reload/pom.xml:73]:
```suggestion
</project>

```

File: flow-tests/test-ccdm-flow-navigation/pom-production.xml [hunk 7]
```diff
@@ -58,47 +56,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-ccdm-flow-navigation/pom-production.xml:59]:
```suggestion
</project>

```

File: flow-tests/test-ccdm-flow-navigation/pom.xml [hunk 8]
```diff
@@ -54,47 +52,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-ccdm-flow-navigation/pom.xml:55]:
```suggestion
</project>

```

File: flow-tests/test-ccdm/pom-production.xml [hunk 9]
```diff
@@ -66,47 +64,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-ccdm/pom-production.xml:67]:
```suggestion
</project>

```

File: flow-tests/test-custom-route-registry/pom.xml [hunk 10]
```diff
@@ -53,37 +51,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-custom-route-registry/pom.xml:54]:
```suggestion
</project>

```

File: flow-tests/test-dev-mode/pom.xml [hunk 11]
```diff
@@ -65,38 +64,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-dev-mode/pom.xml:67]:
```suggestion
</project>

```

File: flow-tests/test-embedding/test-embedding-application-theme/pom.xml [hunk 12]
```diff
@@ -70,37 +68,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-embedding/test-embedding-application-theme/pom.xml:71]:
```suggestion
</project>

```

File: flow-tests/test-embedding/test-embedding-generic/pom.xml [hunk 13]
```diff
@@ -80,37 +77,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-embedding/test-embedding-generic/pom.xml:80]:
```suggestion
</project>

```

File: flow-tests/test-embedding/test-embedding-production-mode/pom.xml [hunk 14]
```diff
@@ -78,37 +76,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-embedding/test-embedding-production-mode/pom.xml:79]:
```suggestion
</project>

```

File: flow-tests/test-embedding/test-embedding-reusable-theme/pom.xml [hunk 15]
```diff
@@ -74,37 +72,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-embedding/test-embedding-reusable-theme/pom.xml:75]:
```suggestion
</project>

```

File: flow-tests/test-embedding/test-embedding-theme-variant/pom.xml [hunk 16]
```diff
@@ -62,38 +61,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-embedding/test-embedding-theme-variant/pom.xml:64]:
```suggestion
</project>

```

File: flow-tests/test-frontend/test-npm/pom-production.xml [hunk 17]
```diff
@@ -83,37 +81,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-frontend/test-npm/pom-production.xml:84]:
```suggestion
</project>

```

File: flow-tests/test-frontend/test-npm/pom.xml [hunk 18]
```diff
@@ -79,37 +77,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-frontend/test-npm/pom.xml:80]:
```suggestion
</project>

```

File: flow-tests/test-frontend/test-pnpm/pom-production.xml [hunk 19]
```diff
@@ -126,36 +124,5 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
     </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-frontend/test-pnpm/pom-production.xml:128]:
```suggestion
</project>

```

File: flow-tests/test-frontend/test-pnpm/pom.xml [hunk 20]
```diff
@@ -78,37 +76,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-frontend/test-pnpm/pom.xml:79]:
```suggestion
</project>

```

File: flow-tests/test-fusion-csrf-context/pom.xml [hunk 21]
```diff
@@ -98,47 +96,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-fusion-csrf-context/pom.xml:99]:
```suggestion
</project>

```

File: flow-tests/test-live-reload/pom.xml [hunk 22]
```diff
@@ -70,37 +68,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-live-reload/pom.xml:71]:
```suggestion
</project>

```

File: flow-tests/test-memory-leaks/pom.xml [hunk 23]
```diff
@@ -60,37 +59,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-memory-leaks/pom.xml:62]:
```suggestion
</project>

```

File: flow-tests/test-misc/pom.xml [hunk 24]
```diff
@@ -60,37 +58,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-misc/pom.xml:61]:
```suggestion
</project>

```

File: flow-tests/test-multi-war/deployment/pom.xml [hunk 25]
```diff
@@ -46,37 +44,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-multi-war/deployment/pom.xml:47]:
```suggestion
</project>

```

File: flow-tests/test-no-theme/pom.xml [hunk 26]
```diff
@@ -93,38 +92,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-no-theme/pom.xml:95]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-devmode.xml [hunk 27]
```diff
@@ -89,47 +87,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-devmode.xml:90]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-prod-fallback.xml [hunk 28]
```diff
@@ -99,47 +97,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-prod-fallback.xml:100]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-production.xml [hunk 29]
```diff
@@ -99,47 +97,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-production.xml:100]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-custom-frontend-directory/pom.xml [hunk 30]
```diff
@@ -91,47 +89,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-custom-frontend-directory/pom.xml:92]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-general/pom.xml [hunk 31]
```diff
@@ -110,47 +108,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-general/pom.xml:111]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-no-buildmojo/pom.xml [hunk 32]
```diff
@@ -100,48 +98,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-no-buildmojo/pom.xml:101]:
```suggestion
</project>

```

File: flow-tests/test-npm-only-features/test-npm-performance-regression/pom.xml [hunk 33]
```diff
@@ -138,47 +136,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-npm-only-features/test-npm-performance-regression/pom.xml:139]:
```suggestion
</project>

```

File: flow-tests/test-pwa/pom-production.xml [hunk 34]
```diff
@@ -77,37 +75,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-pwa/pom-production.xml:78]:
```suggestion
</project>

```

File: flow-tests/test-pwa/pom.xml [hunk 35]
```diff
@@ -73,37 +71,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-pwa/pom.xml:74]:
```suggestion
</project>

```

File: flow-tests/test-redeployment/pom.xml [hunk 36]
```diff
@@ -76,37 +76,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-redeployment/pom.xml:79]:
```suggestion
</project>

```

File: flow-tests/test-root-context/pom.xml [hunk 37]
```diff
@@ -232,4 +189,4 @@
             </build>
         </profile>
     </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-root-context/pom.xml:192]:
```suggestion
</project>

```

File: flow-tests/test-root-ui-context/pom.xml [hunk 38]
```diff
@@ -71,37 +70,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-root-ui-context/pom.xml:73]:
```suggestion
</project>

```

File: flow-tests/test-router-custom-context/pom.xml [hunk 39]
```diff
@@ -68,38 +67,5 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
 
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-router-custom-context/pom.xml:71]:
```suggestion
</project>

```

File: flow-tests/test-servlet/pom.xml [hunk 40]
```diff
@@ -62,37 +61,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-servlet/pom.xml:64]:
```suggestion
</project>

```

File: flow-tests/test-theme-no-polymer/pom.xml [hunk 41]
```diff
@@ -57,38 +56,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-theme-no-polymer/pom.xml:59]:
```suggestion
</project>

```

File: flow-tests/test-themes/pom.xml [hunk 42]
```diff
@@ -56,38 +55,4 @@
...
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
-
+</project>
```
Comment (caalador) [inline flow-tests/test-themes/pom.xml:58]:
```suggestion
</project>

```

File: flow-tests/test-v14-bootstrap/pom-production.xml [hunk 43]
```diff
@@ -71,37 +69,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-v14-bootstrap/pom-production.xml:72]:
```suggestion
</project>

```

File: flow-tests/test-v14-bootstrap/pom.xml [hunk 44]
```diff
@@ -67,37 +65,4 @@
...
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
-</project>
+</project>
```
Comment (caalador) [inline flow-tests/test-v14-bootstrap/pom.xml:68]:
```suggestion
</project>

```

**Review by caalador — CHANGES_REQUESTED** [review]
One direct issue is that this will break all the add-on local builds as they depend on the drivers.xml defined in flow/master

File: flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java [hunk 45]
```diff
@@ -59,93 +47,23 @@ private ChromeDriverLocator() {
      *             method
      */
     static void fillEnvironmentProperty() {
```
Comment (mcollovati) [inline flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java:49]: Javadoc for `fillEnvironmentProperty` states that an `UncheckedIOException` may be thrown "on io exceptions of the find(Path, int, BiPredicate, FileVisitOption...) method" . Is this still valid or may we remove the `@throws` annotation?

File: flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java (hunk 1 above)
Comment (Artur-) [reply flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java]: Simplified

File: flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java (hunk 45 above)
Comment (Artur-) [reply flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java:44]: Removed

File: flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java [hunk 46]
```diff
@@ -17,135 +17,46 @@
...
-        List<Path> driverPaths = getDriverPaths(driverDirectory);
-
-        if (driverPaths.isEmpty()) {
-            System.out.println("No " + CHROMEDRIVER_NAME_PART + " found at \""
-                    + driverDirectory.toAbsolutePath() + "\"\n"
-                    + "  Verify that the path is correct and that driver-binary-downloader-maven-plugin has been run at least once.");
-            return null;
+            // This sets the same property
+            WebDriverManager.chromedriver().setup();
+        } else if (chromedriverProperty != null) {
```
Comment (mcollovati) [inline flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java:55]: IntelliJ IDEA suggests this: "Condition 'chromedriverProperty != null' is always 'true'"

Comment (caalador) [reply flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java:55]: That's actually true as if it is null we will use the first if.

Comment (Artur-) [reply flow-test-util/src/main/java/com/vaadin/flow/testutil/ChromeDriverLocator.java]: Simplified, thanks

**Review by caalador — COMMENTED** [review]
Apart from the removal of drivers.xml this would seem ok.

**Comment by Artur-** [comment]
So should I put it back? It will be outdated quite quickly

**Review by caalador — APPROVED** [review]
It can be removed when we get all places that depend on it updated. And also older versions use the file for drivers when executing tests locally.

CDI, Spring and Quarkus for instance seem to have their own and those are both outdated.

https://github.com/vaadin/flow-starters and MPR seem to be the only active ones left using the flow drivers.xml

**Review by mcollovati — APPROVED** [review]

**Pull request merged**